operator/v1/types_network.go

```diff
@@ -507,6 +507,65 @@ type GatewayConfig struct {
+	// configuration is used. Check individual members fields within ipv6 for details of default values.
+	// +optional
+	IPv6 *IPv6GatewayConfig `json:"ipv6,omitempty"`
+}
+
+type IPv4GatewayConfig struct {
+	// v4InternalMasqueradeSubnet contains the v4 masquerade addresses used internally by
+	// ovn-kubernetes to enable host to service traffic. Each host in the cluster is configured with these
+	// addresses, as well as the shared gateway bridge interface. The values can be changed after
+	// installation. The subnet chosen should not overlap with other networks specified for
+	// OVN-Kubernetes as well as other networks used on the host. Additionally the subnet must
+	// be large enough to accommodate 6 IPs (maximum prefix length /29).
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.
+	// The current default subnet is 169.254.169.0/29
+	// +kubebuilder:validation:MaxLength=18
+	// +kubebuilder:validation:XValidation:rule="self.indexOf('/') == self.lastIndexOf('/')",message="CIDR format must contain exactly one '/'"
+	// +kubebuilder:validation:XValidation:rule="[int(self.split('/')[1])].all(x, x <= 29 && x >= 0)",message="subnet must be in the range /0 to /29 inclusive"
+	// +kubebuilder:validation:XValidation:rule="self.split('/')[0].split('.').size() == 4",message="a valid IPv4 address must contain 4 octets"
+	// +kubebuilder:validation:XValidation:rule="[self.findAll('[0-9]+')[0]].all(x, x != '0' && int(x) <= 255 && !x.startsWith('0'))",message="first IP address octet must not contain leading zeros, must be greater than 0 and less or equal to 255"
+	// +kubebuilder:validation:XValidation:rule="[self.findAll('[0-9]+')[1], self.findAll('[0-9]+')[2], self.findAll('[0-9]+')[3]].all(x, int(x) <= 255 && (x == '0' || !x.startsWith('0')))",message="IP address octets must not contain leading zeros, and must be less or equal to 255"
+	// +optional
+	// + ---
+	// + The regex for the IPV4 CIDR range was taken from other feilds in the Openshift API
+	// + The value must be in IPV4 CIDR format
+	V4InternalMasqueradeSubnet string `json:"v4InternalMasqueradeSubnet,omitempty"`
+}
+
+type IPv6GatewayConfig struct {
+	// v6InternalMasqueradeSubnet contains the v6 masquerade addresses used internally by
+	// ovn-kubernetes to enable host to service traffic. Each host in the cluster is configured with these
+	// addresses, as well as the shared gateway bridge interface. The values can be changed after
+	// installation. The subnet chosen should not overlap with other networks specified for
+	// OVN-Kubernetes as well as other networks used on the host. Additionally the subnet must
+	// be large enough to accommodate 6 IPs (maximum prefix length /125).
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.
```

Comment:
I think here we need to mention dual addresses are forbidden
```suggestion
	// Note that IPv6 dual addresses are not permitted.
	// When omitted, this means no opinion and the platform is left to choose a reasonable default which is subject to change over time.
```

We also need a new validation for this
```
// +kubebuilder:validation:XValidation:rule="!self.contains('.')",message="IPv6 dual addresses are not permitted, value should not contain `.` characters"
```